agenda/presentation/src/main/java/com/pronoidsoftware/agenda/presentation/detail/components/event/photo/EventDetailPhotoHolder.kt

```diff
@@ -0,0 +1,136 @@
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.geometry.CornerRadius
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.drawscope.Stroke
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import coil.compose.AsyncImage
+import com.pronoidsoftware.agenda.presentation.R
+import com.pronoidsoftware.core.presentation.designsystem.PlusIcon
+import com.pronoidsoftware.core.presentation.designsystem.TaskyLightBlue2
+import com.pronoidsoftware.core.presentation.designsystem.TaskyTheme
+
+@Composable
+private fun PhotoBorder(
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    color: Color = TaskyLightBlue2,
+    size: Dp = 60.dp,
+    strokeWidth: Dp = 2.dp,
+    cornerRadius: Dp = 5.dp,
+    content: @Composable () -> Unit = { },
+) {
+    val density = LocalDensity.current
+    val cornerRadiusPx = with(density) { cornerRadius.toPx() }
+    Box(
+        modifier = modifier
+            .size(size + strokeWidth)
+            .clip(RoundedCornerShape(cornerRadius))
+            .clickable {
+                onClick()
+            },
+        contentAlignment = Alignment.Center,
+    ) {
+        content()
+        Canvas(
+            modifier = Modifier.fillMaxSize(),
+        ) {
+            drawRoundRect(
+                color = color,
+                cornerRadius = CornerRadius(cornerRadiusPx, cornerRadiusPx),
+                style = Stroke(
+                    width = with(density) { strokeWidth.toPx() },
+                ),
+            )
+        }
+    }
+}
+
+@Composable
+fun AddPhotoButton(
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    addIcon: ImageVector = PlusIcon,
+    iconColor: Color = TaskyLightBlue2,
+) {
+    PhotoBorder(
+        onClick = onClick,
+        modifier = modifier,
+    ) {
+        Icon(
+            imageVector = addIcon,
+            contentDescription = stringResource(id = R.string.add_photos),
+            tint = iconColor,
+        )
+    }
+}
+
+@Composable
+fun PhotoThumbnail(photoId: PhotoId, onClick: () -> Unit, modifier: Modifier = Modifier) {
+    PhotoBorder(
+        modifier = modifier,
+        onClick = onClick,
+    ) {
+        when (photoId) {
+            is PhotoId.PhotoUri -> {
+                AsyncImage(
+                    model = photoId.uri,
```

Comment:
I would've thought that `PhotoId` is a typealias or value class for photo IDs, but it seems it's rather a data structure for photos in general. How comes you chose to give it this name rather than just `Photo` or so?

Reply:
it's just a wrapper for either a URI or a resID since I wanted to be able to pass in either without having to duplicate too much code. I'd prefer to not have to worry about res ID at all, but I couldn't find a way to adequately preview these photo composables without using `Image` instead of Coil's `AsyncImage` which can accept resource IDs in previews